File: backend/src/main/java/chatogether/ChaTogether/persistence/Stats.java
package chatogether.ChaTogether.persistence;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "stats")
public class Stats {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(name = "month_value")
    private Integer month;

    @Column(name = "year_value")
    private Integer year;
    private Integer newUsersCount = 0;
    private Integer videoRoomsCount = 0;
    private Integer groupChatsCount = 0;
    private Integer privateChatsCount = 0;
}

File: frontend/lib/interfaces/stats.dart
class Stats {
  final int id;
  final int month;
  final int year;
  final int newUsersCount;
  final int videoRoomsCount;
  final int groupChatsCount;
  final int privateChatsCount;

  Stats({
    required this.id,
    required this.month,
    required this.year,
    required this.newUsersCount,
    required this.videoRoomsCount,
    required this.groupChatsCount,
    required this.privateChatsCount,
  });

  factory Stats.fromJson(Map<String, dynamic> json) {
    return Stats(
      id: json['id'],
      month: json['month'],
      year: json['year'],
      newUsersCount: json['newUsersCount'],
      videoRoomsCount: json['videoRoomsCount'],
      groupChatsCount: json['groupChatsCount'],
      privateChatsCount: json['privateChatsCount'],
    );
  }
}
